### ISSUE TITLE
Reported "Supported shields" list includes boards

### ISSUE THREAD

**Issue by erwango** [zephyrproject-rtos/zephyr#26522]
**Describe the bug**
When west reports the list of supported boards and shields the "Supported shields" list is wrong:

```
Supported Shields:

 adafruit_2_8_tft_touch_v2
 boostxl_ulpsense
 dfrobot_can_bus_v2_0
 disco_l475_iot1                         -> Not a shield
 frdm_k64f                                  -> Not a shield
 sam4e_xpro
 esp_8266
 frdm_cr20a
 frdm_kw41z
 link_board_eth
 lmp90100_evb
 mikroe_adc_click
 mikroe_eth_click
 nrf52840dk_nrf52840
 sparkfun_sara_r4
 reel_board                            -> Not a shield
 sh1106_128x64
 ssd1306_128x32
 ssd1306_128x64
 st7789v_tl019fqv01
 st7789v_waveshare_240x240
 waveshare_epaper_gdeh0154a07
 waveshare_epaper_gdeh0213b1
 waveshare_epaper_gdeh0213b72
 waveshare_epaper_gdeh029a1
 waveshare_epaper_gdew075t7
 frdm_k64f                   -> Not a shield
 nrf52840dk_nrf52840
 wnc_m14a2a
 stm32mp157c_dk2    -> Not a shield
 x_nucleo_idb05a1
 x_nucleo_iks01a1
 stm32mp157c_dk2
 x_nucleo_iks01a2
 stm32mp157c_dk2
 stm32mp157c_dk2
 x_nucleo_iks01a3
 x_nucleo_iks01a3_shub
 nucleo_f411re              -> Not a shield
 x_nucleo_iks02a1
 x_nucleo_iks02a1_mic
 x_nucleo_iks02a1_shub
```

It seems that this list includes the boards available under `boards/shields/*/boards/*`

**To Reproduce**
`west build -b nucleo_f401re -t usage`

**Expected behavior**
List should only include shields

**Comment by mbolivar-nordic** [comment]
@erwango This appears to be due to the use of GLOB_RECURSE here:

https://github.com/zephyrproject-rtos/zephyr/blob/4e39938942a5beb7d4eef79855e45af065c637c8/cmake/app/boilerplate.cmake#L326

The problem seems to be that the RECURSE part makes CMake pick up files like boards/shields/esp_8266/boards/disco_l475_iot1.overlay, which then show up in the output.

I found the use of GLOB_RECURSE was originally introduced in a commit you authored:

6af514b67f8bb6e6a82c70fae2d98415202cb3be

I don't know the directory layout well enough to diagnose further because I don't use shields, but perhaps this line can use GLOB instead of GLOB_RECURSE?

https://cmake.org/cmake/help/latest/command/file.html#glob

**Comment by mbolivar-nordic** [comment]
Clearing my assignment since this does not appear to be a west issue.

**Comment by erwango** [comment]
Thanks @mbolivar-nordic I'll get deeper into it.

**Comment by nashif** [comment]
This has changed in the tree, I do not see "Supported shields" anymore, just boards, so closing.

**Comment by tejlmand** [comment]
was fixed as part of this: https://github.com/zephyrproject-rtos/zephyr/pull/30572

**Comment by erwango** [comment]
Reopening:
```
$ west build -b nucleo_f401re -t shields

adafruit_2_8_tft_touch_v2
adafruit_winc1500
atmel_rf2xx
atmel_rf2xx_arduino
atmel_rf2xx_legacy
atmel_rf2xx_mikrobus
atmel_rf2xx_xplained
atmel_rf2xx_xpro
boostxl_ulpsense
buydisplay_2_8_tft_touch_arduino
buydisplay_3_5_tft_touch_arduino
dac80508_evm
dfrobot_can_bus_v2_0
esp_8266
esp_8266_arduino
esp_8266_mikrobus
frdm_cr20a
frdm_k64f
frdm_k64f
frdm_kw41z
inventek_eswifi
inventek_eswifi_arduino_spi
inventek_eswifi_arduino_uart
link_board_eth
lmp90100_evb
lpcxpresso55s16_ns
lpcxpresso55s69_cpu0
lpcxpresso55s69_ns
ls013b7dh03
mikroe_adc_click
mikroe_eth_click
nrf52840dk_nrf52840
nrf52840dk_nrf52840
nrf52840dk_nrf52840
nrf52840dk_nrf52840
nucleo_f411re
reel_board
sam4e_xpro
sh1106_128x64
sparkfun_sara_r4
ssd1306_128x32
ssd1306_128x64
ssd1306_128x64_spi
st7789v_tl019fqv01
st7789v_waveshare_240x240
stm32mp157c_dk2
stm32mp157c_dk2
stm32mp157c_dk2
stm32mp157c_dk2
v2c_daplink
v2c_daplink_cfg
waveshare_epaper_gdeh0154a07
waveshare_epaper_gdeh0213b1
waveshare_epaper_gdeh0213b72
waveshare_epaper_gdeh029a1
waveshare_epaper_gdew075t7
wnc_m14a2a
x_nucleo_idb05a1
x_nucleo_iks01a1
x_nucleo_iks01a2
x_nucleo_iks01a2_shub
x_nucleo_iks01a3
x_nucleo_iks01a3_shub
x_nucleo_iks02a1
x_nucleo_iks02a1_mic
x_nucleo_iks02a1_shub
```

In this list can be found board names: `frdm_k64f`, `reel_board`, `nucleo_f411re`, ...
Basically, I think that we're getting all boards that are in `shields/*/boards/`.

**Comment by mbolivar-nordic** [comment]
Right. #30572 didn't change anything with regard to how the list of shields is populated, so it has no effect on this issue.

**Comment by tejlmand** [comment]
> Right. #30572 didn't change anything with regard to how the list of shields is populated, so it has no effect on this issue.

Seems you're right, my bad for not noticing that this wasn't fixed by your cleanup.

This issue is fixed here: #31877
